timesketch/lib/analyzers/yetiindicators.py

```diff
@@ -91,41 +94,38 @@ def run(self):
         if not self.yeti_api_root or not self.yeti_api_key:
             return 'No Yeti configuration settings found, aborting.'
 
-        self.get_intrusion_sets()
-        actors_found = []
-        for intrusion_set in self.intel.values():
-            if not intrusion_set['indicators']:
-                continue
-
-            found = False
-
-            for indicator in intrusion_set['indicators']:
-                query = build_query_for_indicators([indicator])
-
-                events = self.event_stream(query_string=query,
-                                           return_fields=[])
-
-                name = intrusion_set['name']
-                for event in events:
-                    found = True
-                    event.add_emojis([emojis.get_emoji('SKULL')])
-                    event.add_tags([name])
-                    event.commit()
-                    event.add_comment(
-                        'Indicator "{0:s}" found for actor "{1:s}"'.format(
-                            indicator['name'], name))
-
-            if found:
-                actors_found.append(name)
-                self.sketch.add_view(
-                    'Domain activity for actor {0:s}'.format(name),
-                    self.NAME,
-                    query_string=query)
-
-        if actors_found:
-            return '{0:d} actors were found! [{1:s}]'.format(
-                len(actors_found), ', '.join(actors_found))
-        return 'No indicators were found in the timeline.'
+        self.get_indicators('x-regex')
+
+        entities_found = set()
+
+        events = self.event_stream(query_string='*',
+                                   return_fields=['message'])
+        total_matches = 0
+        matching_indicators = set()
+        for event in events:
+            for _id, indicator in self.intel.items():
+                regexp = indicator['compiled_regexp']
+                if regexp.search(event.source['message']):
```

Comment:
This makes sense. This is only for 'regex' type of indicators which are super generic and can be applied to almost anything. I picked the message source, because it's what humans read and therefore probably what people would base themselves on to write the regexes.

Reply:
SG